### PR TITLE
Remove unused dBFT APIs

### LINKED ISSUE #84: Refactor dBFT APIs
## Is your feature request related to a problem? Please describe.
Some of dBFT APIs are not used since `WithNewBlockFromContext` addition. E.g. `WithGetConsensusAddress` always returns an empty address in NeoGo.

## Describe the solution you'd like
We need to review all unused dBFT APIs and remove them.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
This PR removes exported APIs from interfaces declared in the dbft package. Those APIs are remove that are not used by dBFT. If dBFT user wants to access some of removed dbft payloads APIs, then it should cast dbft payload interface to the corresponding locally-defined payload implementation.

Regarding to the rest of APIs, @roman-khimov, need your opinion: 
* `Timer` related Config API is currently unused in both NeoGo and Bane (`WithTimer`, `WithTimestampIncrement`). Default timer implementation is used, thus, timer API may be removed (except `WithSecondsPerBlock`). However, in general it's nice to have the ability to customize timer. What is the desired behaviour? (I vote to keep it).
* `VerifyPrepareResponse` is currently unused by NeoGo and Bane, no verification is being performed, but it may be useful for future Bane consensus, thus I'd suggest to keep it.

Close #84.

### CODE REVIEW

File: dbft.go [hunk 1]
```diff
@@ -370,15 +369,6 @@ func (d *DBFT[H, A]) processMissingTx() {
 // with it, it sends a changeView request and returns false. It's only valid to
 // call it when all transactions for this block are already collected.
 func (d *DBFT[H, A]) createAndCheckBlock() bool {
```
Comment (roman-khimov) [inline dbft.go:371]: Likely it's not needed even here.

File: internal/block/block.go [hunk 2]
```diff
@@ -72,12 +67,12 @@ func (b *neoBlock) SetTransactions(txx []dbft.Transaction[crypto.Uint256]) {
...
 
 // NewBlock returns new block.
-func NewBlock(timestamp uint64, index uint32, prevHash crypto.Uint256, version uint32, nonce uint64, txHashes []crypto.Uint256) dbft.Block[crypto.Uint256, crypto.Uint160] {
+func NewBlock(timestamp uint64, index uint32, prevHash crypto.Uint256, nonce uint64, txHashes []crypto.Uint256) dbft.Block[crypto.Uint256, crypto.Uint160] {
 	block := new(neoBlock)
 	block.base.Timestamp = uint32(timestamp / 1000000000)
 	block.base.Index = index
 	block.base.PrevHash = prevHash
-	block.base.Version = version
+	block.base.Version = 0 // Not a consensus subject.
```
Comment (roman-khimov) [inline internal/block/block.go:64]: It's a consensus subject, but it's implementation-specific. We will have a new block version after https://github.com/neo-project/neo/issues/1526, but dBFT logic doesn't care much, it's all about particular proposal verification.

File: internal/consensus/block.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ type (
 	// base is a structure containing all
 	// hashable and signable fields of the block.
 	base struct {
-		ConsensusData uint64
+		ConsensusData uint64 //nolint:staticcheck this field is unused in the current example, but it likely to be used in some real implementation.
```
Comment (roman-khimov) [inline internal/consensus/block.go:16]: NextConsensus can be kept as well then.

File: recovery_request.go [hunk 4]
```diff
@@ -1,7 +1,4 @@
 package dbft
 
 // RecoveryRequest represents dBFT RecoveryRequest message.
-type RecoveryRequest interface {
-	// Timestamp returns this message's timestamp.
-	Timestamp() uint64
-}
+type RecoveryRequest interface{}
```
Comment (roman-khimov) [inline recovery_request.go:4]: `any`? It's somewhat strange then though.

**Review by roman-khimov — CHANGES_REQUESTED** [review]
I'd move `timer.Timer` to `dbft.Timer` (the interface itself) and then provide an implementation in `timer`. `VerifyPrepareResponse` is OK to keep.

File: block.go [hunk 5]
```diff
@@ -16,8 +16,6 @@ type Block[H Hash, A Address] interface {
 	Index() uint32
 	// ConsensusData is a random nonce.
 	ConsensusData() uint64
-	// NextConsensus returns hash of the validators of the next block.
-	NextConsensus() A
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply block.go:20]: Fixed.

File: config.go [hunk 6]
```diff
@@ -57,12 +57,10 @@ type Config[H Hash, A Address] struct {
 	// list of the validators of the next block.
 	// If this function ever returns 0-length slice, dbft will panic.
 	GetValidators func(...Transaction[H]) []PublicKey
-	// GetConsensusAddress returns hash of the validator list.
-	GetConsensusAddress func(...PublicKey) A
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply config.go:61]: Fixed.

File: consensus_message.go [hunk 7]
```diff
@@ -12,7 +12,7 @@ type ConsensusMessage[H Hash, A Address] interface {
 	// GetChangeView returns payload as if it was ChangeView.
 	GetChangeView() ChangeView
 	// GetPrepareRequest returns payload as if it was PrepareRequest.
-	GetPrepareRequest() PrepareRequest[H, A]
+	GetPrepareRequest() PrepareRequest[H]
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply consensus_message.go:15]: Fixed.

File: context.go [hunk 8]
```diff
@@ -40,8 +40,6 @@ type Context[H Hash, A Address] struct {
 	PrimaryIndex uint
 	Version      uint32
 
-	// NextConsensus is a hash of the validators which will be accepting the next block.
-	NextConsensus A
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply context.go:44]: Fixed.

File: dbft.go (hunk 1 above)
Comment (AnnaShaleva) [reply dbft.go]: Fixed.

File: internal/block/block.go (hunk 2 above)
Comment (AnnaShaleva) [reply internal/block/block.go]: Comment is fixed.

> We will have a new block version after https://github.com/neo-project/neo/issues/1526

Sure, I'm tracking this issue.

File: internal/consensus/block.go (hunk 3 above)
Comment (AnnaShaleva) [reply internal/consensus/block.go]: Fixed.

File: recovery_request.go (hunk 4 above)
Comment (AnnaShaleva) [reply recovery_request.go]: I've removed this commit, `Timestamp()` doesn't hurt here.

File: timer.go [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package dbft
+
+import (
+	"time"
+
+	timer2 "github.com/nspcc-dev/dbft/timer"
```
Comment (roman-khimov) [inline timer.go:6]: `timer2`?

File: timer.go [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+// Timer is an interface which implements all time-related
+// functions. It can be mocked for testing.
+type Timer interface {
+	// Now returns current time.
+	Now() time.Time
+	// Reset
+	Reset(s timer2.HV, d time.Duration)
```
Comment (roman-khimov) [inline timer.go:15]: I think it'd be inconvenient to use without `dbft.HeightView`

Comment (AnnaShaleva) [reply timer.go]: Fixed!

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**